gpii/node_modules/eventLog/test/EventLogTests.js

```diff
@@ -0,0 +1,339 @@
+    var logFile = os.tmpdir() + "/gpii-test-eventLog-" + Date.now();
+    teardowns.push(function () {
+        fs.unlinkSync(logFile);
+    });
+
+    //gpii.eventLog.setInstallId(gpii.tests.eventLog.installID);
+
+    // Mock getTimestamp function
+    var oldGetTimestamp = gpii.eventLog.getTimestamp;
+    teardowns.push(function () {
+        // Restore getTimestamp.
+        gpii.eventLog.getTimestamp = oldGetTimestamp;
+    });
+    gpii.eventLog.getTimestamp = function () {
+        return gpii.tests.eventLog.timestamp;
+    };
+
+    return logFile;
+};
+
+/**
+ * Checks a log line.
+ *
+ * @param line {String} The log lone.
+ * @param expected {Object} What it should parse into.
+ */
+gpii.tests.eventLog.checkLogLine = function (line, expected) {
+    jqUnit.expect(3);
+    jqUnit.assertTrue("Line should look like a JSON object", !!line.match(/^{.*}$/));
+    var obj = JSON.parse(line);
+    jqUnit.assertEquals("JSON should be an object", "object", typeof(obj));
+
+    // data.duration is tricky to predict precisely, so just make sure it parses to a time within 5 seconds.
+    if (obj.data && obj.data.hasOwnProperty("duration")) {
+        jqUnit.expect(2);
+        jqUnit.assertTrue("data.duration property should be >= 0", obj.data.duration >= 0);
+        jqUnit.assertTrue("data.duration property shouldn't be more than 5 seconds", obj.data.duration < 5);
+        obj.data.duration = 0;
+    }
+
+    jqUnit.assertDeepEq("Parsed log should match expected", expected, obj);
+};
+
+/**
+ * Checks an entire log file.
+ *
+ * @param logFile {String} The log file
+ * @param expected {Object[]} Array of objects expected for each line.
+ * @return {Promise} A promise that resolves when the file is read.
+ */
+gpii.tests.eventLog.checkLogFile = function (logFile, expected) {
+    var promise = fluid.promise();
+    var reader = readline.createInterface({
+        input: fs.createReadStream(logFile)
+    });
+
+    // Inspect each line of the log, making sure each one parses into the expected object.
+    var lineNumber = 0;
+    reader.on("line", function (line) {
+        console.log("Log line", lineNumber, line);
+
+        if (lineNumber >= expected.length) {
+            jqUnit.fail("Log file should not contain more data.");
+            promise.reject();
+            reader.close();
+            return;
+        }
+        gpii.tests.eventLog.checkLogLine(line, expected[lineNumber]);
+        lineNumber++;
+    });
+
+    reader.on("close", function () {
+        if (!promise.disposition) {
+            promise.resolve();
+        }
+    });
+
+    return promise;
+};
+
+jqUnit.asyncTest("Uncaught exception test", function () {
+    jqUnit.expect(1);
+
+    var logFile = gpii.tests.eventLog.prepareLogFile();
+    gpii.tests.eventLog.createEventLogComponent(logFile);
+    // Clear the log file
+    fs.unlinkSync(logFile);
+
+    // Unable to perform a "real" test of logging uncaught exceptions, so just check if the listener has been added.
```

Comment:
Well, actually you could, but I won't ask you to : P
https://github.com/GPII/universal/blob/master/tests/JournalIntegrationTests.js#L427